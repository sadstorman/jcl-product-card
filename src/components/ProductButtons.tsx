import { ProductContext } from './ProductCard';
import React, { CSSProperties, useCallback, useContext } from 'react';
import styles from "../styles/styles.module.css";

export interface Props {
    className?: string;
    style?: CSSProperties
}

export const ProductButtons = ({ className, style }: Props) => {

    // Extraer mas count del context
    const { counter, increaseBy, maxCount } = useContext(ProductContext)

    // isMaxReached = useCallback, dependencias [counter, maxCount] 
    // TRUE si el count === maxCount
    // FaLSE si es distinto
    const isMaxReached = useCallback(
        () => !!maxCount && counter === maxCount,
        [counter, maxCount]
    )
    

    return (
        <div className={`${styles.buttonsContainer} ${className} `}
            style={style}
        >
            <button className={styles.buttonMinus} onClick={() => { increaseBy(-1) }} >
                -
            </button>

            <div className={styles.countLabel} >
                {counter}
            </div>

            <button className={`${styles.buttonAdd} ${(isMaxReached()) ? styles.disabled : ''}`} onClick={() => { increaseBy(+1) }}>
                +
            </button>
        </div>
    )
}