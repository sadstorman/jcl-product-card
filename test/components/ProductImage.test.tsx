import React from 'react';
import renderer from 'react-test-renderer'
import { ProductImage, ProductCard } from '../../src/components';
import { producto2 } from './data/products';


describe('ProductTitle', () => {

    test('debe de mostrar el comp correctamente con URL de imagen', () => {
        const wrapper = renderer.create(
            <ProductImage img='imgURL' />
        )

        expect(wrapper.toJSON()).toMatchSnapshot();

    })

    test('debe de mostrar el componente con la imagen del producto', () => {

        const wrapper = renderer.create(
            <ProductCard product={producto2}>
                {
                    () => (
                        <ProductImage />
                    )
                }
            </ProductCard>
        )

        expect(wrapper.toJSON()).toMatchSnapshot();

    })

});
