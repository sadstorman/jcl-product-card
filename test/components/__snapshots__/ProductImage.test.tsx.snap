// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle debe de mostrar el comp correctamente con URL de imagen 1`] = `
<img
  alt="Product image"
  className="productImg undefined"
  src="imgURL"
/>
`;

exports[`ProductTitle debe de mostrar el componente con la imagen del producto 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product image"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;
